src/app/BUILD.gn

```diff
@@ -18,6 +18,7 @@ import("${chip_root}/src/platform/device.gni")
 
 import("${chip_root}/build/chip/buildconfig_header.gni")
 import("common_flags.gni")
+import("icd/icd.gni")
```

Comment:
I see that this is a separate gni.

Could we have a separate BUILD.gn into icd and include things as a dependency (even if source set) instead of flattening things below?

Reply:
Agreed it should be cleaner

Reply:
done